src/flows/neuralspline.jl

```diff
@@ -0,0 +1,235 @@
+"""
+    NeuralSplineCoupling(dim, hdims, K, B, mask_idx, paramtype)
+    NeuralSplineCoupling(dim, K, n_dims_transferred, B, nn, mask)
+
+Neural Rational Quadratic Spline (RQS) coupling bijector [^DBMP2019].
+
+A conditioner network takes the unchanged partition as input and outputs the
+parameters of monotonic rational quadratic splines for the transformed
+coordinates. Batched inputs (matrices with column vectors) are supported.
+
+Arguments
+- `dim::Int`: total input dimension.
+- `hdims::AbstractVector{Int}`: hidden sizes for the conditioner MLP.
+- `K::Int`: number of spline knots per transformed coordinate.
+- `B::AbstractFloat`: boundary/box constraint for spline domain.
+- `mask_idx::AbstractVector{Int}`: indices of the transformed coordinates.
+
+Keyword Arguments
+- `paramtype::Type{<:AbstractFloat}`: parameter element type.
+
+Fields
+- `nn::Flux.Chain`: conditioner that outputs all spline params for all transformed dims.
+- `mask::Bijectors.PartitionMask`: partition specification.
+
+Notes
+- Output dimensionality of the conditioner is `(3K - 1) * n_transformed`.
+- For computation performance, we rely on 
+[`MonotonicSplines.jl`](https://github.com/bat/MonotonicSplines.jl) for the
+building the rational quadratic spline functions.
+- See `MonotonicSplines.rqs_forward` and `MonotonicSplines.rqs_inverse` for forward/inverse 
+and log-determinant computations.
+
+[^DBMP2019]: Durkan, C., Bekasov, A., Murray, I. and Papamarkou, T. (2019). Neural Spline Flows. *NeurIPS.*
+"""
+struct NeuralSplineCoupling{T,A<:Flux.Chain} <: Bijectors.Bijector
+    dim::Int                        # dimension of input
+    K::Int                          # number of knots
+    n_dims_transferred::Int         # number of dimensions that are transformed
+    B::T                            # bound of the knots
+    nn::A                           # networks that parmaterize the knots and derivatives
+    mask::Bijectors.PartitionMask
+end
```

Comment:
`DocStringExtensions` supports docstrings for the struct fields. See [here](https://discourse.julialang.org/t/documenting-elements-of-a-struct/64769/11?u=red-portal)